### ISSUE TITLE
Calling getRootFolder() within any app.php prevents the accepting of federated shares

### ISSUE THREAD

**Issue by paulijar** [owncloud/core#37719]
### Steps to reproduce
1. Have an application installed which has a function call like `$app->getContainer()->getServer()->getRootFolder()` within their `app.php`. The [Music](https://github.com/owncloud/music) app is such, but you can also inject this line of code to the bootstrapping code of another app, and get similar results.
2. Make a federated share from another instance of ownCloud to the first instance (the one with the said app).
3. In the "Shared with you" view, attempt to accept the federated share.

### Expected behaviour
The share can be accepted.

### Actual behaviour
The share is not accepted, but stays in the "Pending" state.

The browser console shows an error
```
Uncaught TypeError: response is null
    responseCallback app.js:236
    jQuery 5
app.js:236:16
```
The owncloud.log has an error:
```
{"reqId":"x8pO5zDRVmmDQPz9tSy7","level":3,"time":"2020-07-20T20:17:32+00:00","remoteAddr":"xxx.xxx.xxx.xxx","user":"xxxx","app":"PHP","method":"POST","url":"\/owncloud_10.4\/ocs\/v1.php\/apps\/files_sharing\/api\/v1\/remote_shares\/pending\/7?format=json","message":"Error: Call to a member function file_exists() on null at \/var\/www\/html\/owncloud_10.4\/apps\/files_sharing\/lib\/Helper.php#285"}
```

This was originally reported at https://github.com/owncloud/music/issues/778. There, it was reported that the bug has been present since ownCloud 10.4.1. Personally, I could reproduce it on 10.4.1.3 but not on 10.2.0.

### Server configuration
**Operating system**: Raspbian GNU/Linux 8.0 (jessie)

**Web server:** nginx/1.6.2

**Database:** sqlite 3.8.7.1

**PHP version:** PHP 7.1.33

**ownCloud version:** 10.4.1.3

**Updated from an older ownCloud or fresh install:** fresh

**Where did you install ownCloud from:** https://download.owncloud.org/community/owncloud-10.4.1.zip (maybe)

**Signing status (ownCloud 9.0 and above):** https://pastebin.com/r9deh0eq (this indicates the modification in the encryption app which I made to reproduce the bug without the Music app)

**The content of config/config.php:** https://pastebin.com/tsCMMT5Y

**List of activated apps:**

```
Enabled:
  - comments: 0.3.0
  - configreport: 0.2.0
  - dav: 0.5.0
  - encryption: 1.4.0
  - federatedfilesharing: 0.5.0
  - federation: 0.1.0
  - files: 1.5.2
  - files_external: 0.7.1
  - files_mediaviewer: 1.0.2
  - files_sharing: 0.12.0
  - files_trashbin: 0.9.1
  - files_versions: 1.3.0
  - firstrunwizard: 1.2.0
  - market: 0.5.0
  - notifications: 0.5.0
  - provisioning_api: 0.5.0
  - systemtags: 0.3.0
  - updatenotification: 0.2.1
Disabled:
  - audioplayer
  - external
  - user_external
```

**Are you using external storage, if yes which one:** no

**Are you using encryption:** no

**Are you using an external user-backend, if yes which one:** no

### Client configuration
**Browser:** Firefox 78.0.2 (64-bit)

**Operating system:** Windows 10

#### Web server error log
```
(no new lines in the log when the error happens)
```

#### ownCloud log (data/owncloud.log)
```
(the related log line already shown above)
```

#### Browser log
```
(the related log line already shown above)
```

**Comment by paulijar** [comment]
So does anyone have any idea what is going on here? Shouldn't it be business-as-usual to call `getRootFolder()` in your app.php? After all, this *is* the documented procedure to connect file system hooks: https://doc.owncloud.com/server/developer_manual/app/fundamentals/hooks.html.

**Comment by cdamken** [comment]
Looks fixed with owncloud/core#37834
